### PR TITLE
Address audit shellcheck issues

### LINKED ISSUE #1660: shellcheck issues
**Detailed Description**
Shellcheck should run against all shell scripts without any errors.

`$ find . -name "*.sh" -type f | grep -v node_modules | grep -v target | xargs shellcheck`

**Actual Behavior**
```
In scripts/deploy/deploy.sh line 4:
cd "$(dirname $0)/.."
              ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
cd "$(dirname "$0")/.."


In scripts/deploy/deploy.sh line 28:
    read -p "Database hostname: " dbHost
    ^--^ SC2162: read without -r will mangle backslashes.


In scripts/deploy/deploy.sh line 29:
    read -p "Database name: " dbName
    ^--^ SC2162: read without -r will mangle backslashes.


In scripts/deploy/deploy.sh line 30:
    read -p "Rosetta user password: " dbPassword
    ^--^ SC2162: read without -r will mangle backslashes.


In scripts/deploy/deploy.sh line 31:
    read -p "Database port: " dbPort
    ^--^ SC2162: read without -r will mangle backslashes.


In scripts/deploy/deploy.sh line 32:
    read -p "Rosetta user: " dbUser
    ^--^ SC2162: read without -r will mangle backslashes.


In scripts/deploy/deploy.sh line 33:
    read -p "Rosetta api port: " apiPort
    ^--^ SC2162: read without -r will mangle backslashes.


In scripts/deploy/deploy.sh line 34:
    sed -e 's/${dbHost}/'"${dbHost}"'/g' -e 's/${dbName}/'"${dbName}"'/g' -e 's/${dbPassword}/'"${dbPassword}"'/g' \
           ^------------^ SC2016: Expressions don't expand in single quotes, use double quotes for that.
                                            ^------------^ SC2016: Expressions don't expand in single quotes, use double quotes for that.
                                                                             ^----------------^ SC2016: Expressions don't expand in single quotes, use double quotes for that.


In scripts/deploy/deploy.sh line 35:
        -e 's/${dbPort}/'"${dbPort}"'/g' -e 's/${dbUser}/'"${dbUser}"'/g' -e 's/${apiPort}/'"${apiPort}"'/g' \
           ^------------^ SC2016: Expressions don't expand in single quotes, use double quotes for that.
                                            ^------------^ SC2016: Expressions don't expand in single quotes, use double quotes for that.
                                                                             ^-------------^ SC2016: Expressions don't expand in single quotes, use double quotes for that.


In scripts/validation/run-validation.sh line 2:
parent_path=$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )
               ^-- SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

Did you mean:
parent_path=$( cd "$(dirname "${BASH_SOURCE[0]}")" || exit ; pwd -P )


In scripts/validation/run-validation.sh line 3:
cd "$parent_path"
^---------------^ SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

Did you mean:
cd "$parent_path" || exit


In scripts/validation/run-validation.sh line 15:
    if ! ./rosetta-cli check:data --configuration-file=./$network/validate-from-genesis.json; then
                                                         ^------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
    if ! ./rosetta-cli check:data --configuration-file=./"$network"/validate-from-genesis.json; then


In scripts/wait-for-mirror-node.sh line 49:
        is_retriable=$(echo $response_body | jq '.retriable')
                            ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean:
        is_retriable=$(echo "$response_body" | jq '.retriable')


In scripts/wait-for-mirror-node.sh line 53:
            echo Request cannot be retried, response body: [$response_body]
                                                           ^--------------^ SC2102: Ranges can only match single chars (mentioned due to duplicates).

For more information:
  https://www.shellcheck.net/wiki/SC2164 -- Use 'cd ... || exit' or 'cd ... |...
  https://www.shellcheck.net/wiki/SC2016 -- Expressions don't expand in singl...
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
```
**Expected Behavior**
No shellcheck errors

**Environment:**
- Version: v0.27.0

**Additional Context**

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Detailed description**:
- Change cli-validation to reduce the number of demo files it has to download
- Fix shellcheck issues
- Fix wait-for-mirror-node.sh retrying indefinitely
- Fix duplicated Rosetta configuration file
- Fix deploying Rosetta binary to a lib directory
- Fix Rosetta CI printing download progress logs

**Which issue(s) this PR fixes**:
Fixes #1660

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: hedera-mirror-rosetta/scripts/deploy/deploy.sh [hunk 1]
```diff
@@ -1,45 +1,45 @@
...
+    cat > "${configfile}" <<EOF
+hedera:
+  mirror:
+    rosetta:
+      db:
+        host: ${dbHost}
+        name: ${dbName}
+        password: ${dbPassword}
+        port: ${dbPort}
+        username: ${dbUser}
```
Comment (Nana-EC) [inline hedera-mirror-rosetta/scripts/deploy/deploy.sh:37]: nit: shouldn't we keep apiPort as a config option?

Comment (steven-sheehy) [reply hedera-mirror-rosetta/scripts/deploy/deploy.sh:37]: It is still a configuration option, is it not?  This input only applies on the very first install, they can always change it and restart. We should only template the minimal options and port has a good default. Besides, not like this deploy file is even used except for our lone dev instance.

Comment (Nana-EC) [reply hedera-mirror-rosetta/scripts/deploy/deploy.sh:37]: Sounds good

**Review by Nana-EC — APPROVED** [review]
LGTM

**Review by ijungmann — APPROVED** [review]

**Pull request merged**